packages/liquid-html-parser/grammar/liquid-html.ohm

```diff
@@ -400,11 +400,16 @@ LiquidDoc <: Helpers {
   openControl:= "@" | end
 
   fallbackNode = "@" anyExceptStar<endOfParam>
-  paramNode = "@param" strictSpace* paramType? strictSpace* paramName (strictSpace* "-")? strictSpace* paramDescription
+  paramNode = "@param" strictSpace* paramType? strictSpace* (optionalParamName | paramName) (strictSpace* "-")? strictSpace* paramDescription
   paramType = "{" strictSpace* paramTypeContent strictSpace* "}"
   paramTypeContent = anyExceptStar<("}"| strictSpace)>
-  paramName = identifierCharacter+
-  paramDescription = anyExceptStar<endOfParam>
+
+  paramName = (~endOfParamName identifierCharacter)+
+  optionalParamName = "[" strictSpace* optionalParamNameContent strictSpace* "]"
+  optionalParamNameContent = anyExceptStar<endOfParamName>
+  endOfParamName = strictSpace* ("]" | "@")
```

Comment:
A param name can end with a `@`? Or is "end" here more like "book ends"?

Reply:
`anyExceptStar<endOfParamName>` translates to `(~endOfParamName any)*`, which will prevent 

`@param [border_width @asdf ]` from matching as a `optionalParamName`. 

Instead it will match `@param [border_width ` and be a `fallbackNode`

Though I'm questioning the direction here 🤔  let's pair!